src/lib/queries/resetRequest.ts

```diff
@@ -0,0 +1,47 @@
+import { knexController } from '../../database/database';
+import { log } from '../utils/log';
+import { redisClient } from '../auth/redis';
+import { v4 as uuid } from 'uuid';
+import { hashPassword } from '../auth/auth';
+import { sendEmail } from '../utils/sendEmail';
+import { PasswordResetRequestBody } from '../../types/auth.types';
+
+export const PASSWORD_RESET_TOKEN_EXPIRE_TIME = 600; // ten minutes
+
+export const handlePasswordResetRequest = async (
+  body: PasswordResetRequestBody
+): Promise<void> => {
+  const userInfo: { id: string; email: string } | undefined =
+    await knexController('user')
+      .where('email', body.email)
+      .where('archived_at', null)
+      .first('id', 'email');
+
+  if (userInfo === undefined) {
+    log.info('password reset request with non existent email');
+    return;
+  }
+
+  const key = `password-reset:${userInfo.id}`;
+
+  const resetTokenCount: number = await redisClient.SCARD(key);
+
+  if (resetTokenCount >= 5) {
+    log.warn('Too many password reset request for user');
+    return;
+  }
+
+  const newResetToken = uuid();
+  const hashedResetToken = await hashPassword(newResetToken);
+
+  await redisClient.SADD(key, [hashedResetToken.hash]);
+
+  // Set expire time if it is not yet set
+  await redisClient.EXPIRE(key, PASSWORD_RESET_TOKEN_EXPIRE_TIME, 'NX');
+
+  await sendEmail({
+    to: userInfo.email,
+    subject: 'Pyysit salasanan vaihtamista',
```

Comment:
```suggestion
    subject: 'Täyttöpaikka: Pyysit salasanan vaihtamista',
```

Reply:
Öhäh, pitäis kattoa miten sendgridissä saa laitettua sen lähettäjä nimen siihen spostiin mukaan

Reply:
Sais lukea siinä otsikossa tai bodyssa imo myös. Helpottaa hakuja jossain säpopalveluissa 

![image](https://github.com/Tampereen-Urheilusukeltajat/Blenderi-Backend/assets/33838340/0333babf-8316-4d4b-a88e-51b8dacf10d1)

Reply:
Nojoo bodyyn kyllä vois laittaa jonkun maininnan.

Reply:
Ja se nimi tähän tyyliin 
![image](https://github.com/Tampereen-Urheilusukeltajat/Blenderi-Backend/assets/45848534/4aa63426-c85b-49c4-a0d5-9958c04b0d6f)

Reply:
Ja sit ehkä Täyttöpaikka eikä yhdistyksen nimeä lähettäjäksi